packages/next/server/web/spec-extension/response.ts

```diff
@@ -44,23 +44,40 @@ export class NextResponse extends Response {
     })
   }
 
-  static redirect(url: string | NextURL | URL, status = 307) {
+  static redirect(
+    url: string | NextURL | URL,
+    status = 307,
+    init: ResponseInit = {}
+  ) {
     if (!REDIRECTS.has(status)) {
       throw new RangeError(
         'Failed to execute "redirect" on "response": Invalid status code'
       )
     }
 
     return new NextResponse(null, {
-      headers: { Location: validateURL(url) },
+      ...init,
+      headers: {
+        ...init.headers,
+        Location: validateURL(url),
+      },
       status,
     })
```

Comment:
`ResponseInit`'s `status` and `statusText` should have higher priority.

Reply:
Redirect have its status argument. so priority will be 2nd argument by my opinion